src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+            case WOLFOS_LIMB_ABDOMEN:
+                bodyPartIndex = 4;
+                break;
+            case WOLFOS_LIMB_TAIL:
+                bodyPartIndex = 5;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_SHIN:
+                bodyPartIndex = 6;
+                break;
+            case 37: // There is no limb with index this large
+                bodyPartIndex = 7;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_PASTERN:
+                bodyPartIndex = 8;
+                break;
+            case WOLFOS_LIMB_BACK_LEFT_PAW:
+                bodyPartIndex = 9;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35E4C.s")
+        if (bodyPartIndex >= 0) {
+            Vec3f bodyPartPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35EE4.s")
+            Matrix_MultVec3f(&bodyPartVec, &bodyPartPos);
+            this->bodyPartsPos[bodyPartIndex].x = bodyPartPos.x;
+            this->bodyPartsPos[bodyPartIndex].y = bodyPartPos.y;
+            this->bodyPartsPos[bodyPartIndex].z = bodyPartPos.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B360E8.s")
+static void* sWolfosEyeTextures[] = { gWolfosNormalEyeOpenTex, gWolfosNormalEyeHalfTex, gWolfosNormalEyeNarrowTex,
+                                      gWolfosNormalEyeHalfTex };
+static void* sWhiteWolfosEyeTextures[] = { gWolfosWhiteEyeOpenTex, gWolfosWhiteEyeHalfTex, gWolfosWhiteEyeNarrowTex,
+                                           gWolfosWhiteEyeHalfTex };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B361A0.s")
+void EnWf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnWf* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36288.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_wf.c", 2157);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36328.s")
+    if ((this->action != WOLFOS_ACTION_WAIT_TO_APPEAR) || !this->isInvisible) {
+        func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B365A8.s")
+        if (this->actor.params == 0) {
+            gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sWolfosEyeTextures[this->eyeIndex]));
+        } else {
+            gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sWhiteWolfosEyeTextures[this->eyeIndex]));
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36740.s")
+        SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                              this->skelAnime.dListCount, EnWf_OverrideLimbDraw, EnWf_PostLimbDraw, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36C8C.s")
+        if (this->fireTimer != 0) {
+            this->actor.colorFilterTimer++;
+            if (1) {}
```

Comment:
Does `globalCtx2` make this not needed?

Reply:
No, it very much seems to want an if (1) there.